File: LLD - Golang/SolidPrinciple/OpenClosedPrinciple/PaymentProcessor.go
package openclosedprinciple

type PaymentProcessorBad struct{}

func (PaymentProcessorBad) Process(paymentType string, amount float64) {
	switch paymentType {
	case "credit":
	// credit card logic
	case "paypal":
		// paypal logic
	}
	_ = amount
}

File: LLD - Java/src/main/java/org/solid/openClosedPrinciple/PaymentProcessor.java
package org.solid.openClosedPrinciple;

public class PaymentProcessor {
    void process(String type, double amount){
        if (type.equals("credit")){
            // credit card logic
        } else if (type.equals("paypal")){
            // paypal logic
        }
        // Adding crypto means modifying the method
    }
}

// Violates Open and Closed principle